src/durableorchestrationclient.ts

```diff
@@ -46,8 +47,14 @@ const URL = url.URL;
  * };
  * ```
  */
-export function getClient(context: unknown): DurableOrchestrationClient {
-    let clientData = getClientData(context as IOrchestrationFunctionContext);
+export function getClient(
+    context: IOrchestrationFunctionContext,
+    clientInputOptions: ClientInput
+): DurableOrchestrationClient {
+    if (!(context instanceof InvocationContext)) {

```

Comment:
If `IOrchestrationFunctionContext` already extends `InvocationContext`, this check seems unnecessary

Similar comment for line 71 `isClientInput` below

Reply:
The reason I added those checks here is because `getClient` is a customer-facing API, so even though the type here is `IOrchestrationFunctionContext` and `ClientInput`, there is no guarantee that the customer actually supplies input with those types

Reply:
I see what you're saying, but I think it's unreasonable for an npm package to validate every single input it gets like this. I'm guessing this is related to this discussion: https://github.com/Azure/azure-functions-nodejs-library/pull/14#discussion_r964309579

I don't have a strict answer on why, but this case just feels excessive to me where the original doesn't. One reason is that the user is manually crafting the return value in the original and it can take many different formats (http response, buffer, string, number, etc.) but in this case they're just passing along a context object already created for them. In other words it's not that they _can_ pass anything in the original it's that there are valid scenarios where they _will_ pass virtually anything.

Another thing is I don't like `instanceof` because it can prevent valid mock testing scenarios

Reply:
Yeah that makes sense, thanks! Do you think I should also remove the check for `isClientInput` or keep it?

Reply:
Probably yeah. Ultimately this isn't unique to the new programming model stuff anyways, so if you decide to do it (which I'm not advocating for) it would be better to do a separate PR and cherry-pick to the other branch

Reply:
Well, it is unique for the programming model because the old one doesn't even have this second `FunctionInput` argument. Regardless, I've remove the validation for both